Resources/Public/Javascript/PageView/PageView.js

```diff
@@ -8,692 +8,348 @@
-        annotationControl = new DlfAnnotationControl(this.map, this.images[0], this.annotationContainers[0]);
-        if (fulltextControl !== undefined) {
-            $(fulltextControl).on("activate-fulltext", $.proxy(annotationControl.deactivate, annotationControl));
-            $(annotationControl).on("activate-annotations", $.proxy(fulltextControl.deactivate, fulltextControl));
+dlfViewer.prototype.getOLControls = function() {
+    if (this.olx.controls === undefined) {
+        this.olx.controls = [];
+        if (this.settings.controls.includes('Attribution')) {
+            this.olx.controls.push(new ol.control.Attribution({
+                collapsed: false,
+                collapsible: true,
+                label: '\u00a9',
+                target: this.settings.controlTargets.Attribution || undefined,
+                tipLabel: this.settings.controlTitles.Attribution || undefined
+            }));
         }
-    }
-    else {
-        $('#tx-dlf-tools-annotations').remove();
-    }
-
-    //
-    // Add image manipulation tool if container is added.
-    //
-    // It is important to know that the image manipulation tool uses a webgl renderer as basis. Therefor the
-    // application has as first to check if the renderer is active. Further it has to check if cors supported through
-    // image.
-    //
-    if ($('#tx-dlf-tools-imagetools').length > 0 && dlfUtils.isWebGLEnabled() && this.isCorsEnabled) {
-
-        // should be called if cors is enabled
-        imageManipulationControl = new dlfViewerImageManipulationControl({
-            controlTarget: $('.tx-dlf-tools-imagetools')[0],
-            layers: dlfUtils.createOl3Layers(images, '*'),
-            map: this.map,
-            view: dlfUtils.createOl3View(images)
-        });
-
-        // bind behavior of both together
-        if (fulltextControl !== undefined) {
-            $(imageManipulationControl).on("activate-imagemanipulation", $.proxy(fulltextControl.deactivate, fulltextControl));
-            $(fulltextControl).on("activate-fulltext", $.proxy(imageManipulationControl.deactivate, imageManipulationControl));
+        if (this.settings.controls.includes('FullScreen')) {
+            this.olx.controls.push(new ol.control.FullScreen({
+                target: this.settings.controlTargets.FullScreen || undefined,
+                tipLabel: this.settings.controlTitles.FullScreen || undefined
+            }));
         }
-        if (annotationControl !== undefined) {
-            $(imageManipulationControl).on("activate-imagemanipulation", $.proxy(annotationControl.deactivate, annotationControl));
-            $(annotationControl).on("activate-annotations", $.proxy(imageManipulationControl.deactivate, imageManipulationControl));
+        if (this.settings.controls.includes('ImageManipulation')) {
+            this.olx.controls.push(new ImageManipulationControl({
+                // autoOpen: false,
```

Comment:
Why this lines are outcommented?

Reply:
For every new map control I gave all possible options, but commented those out, which use the defaults anyway. That way every option is documented, but no unnecessary option parsing happens.

The reason is, that the new map controls can be used without Kitodo.Presentation as stand-alone OpenLayers controls as well. So I thought it would be a good idea to have all options documented. (I am still looking for the best way to release the controls separately without.)

Do you suggest removing the outcommented options?

Reply:
No, they can stay. I just wanted to know why they are there ;)